app/Http/Controllers/Api/ReportbackController.php

```diff
@@ -44,13 +44,28 @@ public function index(Request $request)
             ->where('status', '=', 'approved')
             ->select('posts.id as id', 'signups.campaign_id as campaign_id', 'posts.status as status', 'posts.caption as caption', 'posts.url as url', 'posts.created_at as created_at', 'posts.signup_id as signup_id');
 
-        // Only return Posts for the given campaign_id (if there is one)
         if (isset($request->filter)) {
```

Comment:
I'd be tempted to use `$request->has('filter')` here since this could conflict if Laravel were ever to add a "filter" property to the base request class. Or maybe pull out the array of filters up here and then you can use Laravel's array helpers?

```php
$filters = $request->query('filter');

if (! empty($filters)) {
    if (array_has($filters, 'campaign_id')) {
        $query = $query->where('campaign_id', '=', $filters['campaign_id']);
        // ...
```